frontend/src/pages/dashboard/profile/EditProfile.tsx

```diff
@@ -0,0 +1,178 @@
+  const navigate = useNavigate();
+
+  const { enqueueSnackbar } = useSnackbar();
+
+  let currentUser = useGetIdentity().identity;
+
+  const UserSchema = Yup.object().shape({
+    name: Yup.string()
+      .min(MIN_NAME_LEN, NAME_LEN_ERROR)
+      .max(MAX_NAME_LEN, NAME_LEN_ERROR)
+      .matches(REGEX_NAME, NAME_REGEX_ERROR)
+      .required('Name is required.'),
+    height: Yup.number().positive().integer().min(MIN_HEIGHT).max(MAX_HEIGHT).optional(),
+    reach: Yup.number().positive().integer().min(MIN_REACH).max(MAX_REACH).optional(),
+    pronounId: Yup.number().optional(),
+    highestBoulderingGradeId: Yup.number().optional(),
+    highestTopRopeGradeId: Yup.number().optional(),
+    highestLeadClimbingGradeId: Yup.number().optional(),
+    favouriteGymIds: Yup.array().of(Yup.number()).optional(),
+    sncsCertificationId: Yup.number().optional(),
+    // avatarUrl: Yup.mixed().test('required', 'Avatar is required', (value) => value !== ''),
+  });
+
+  const defaultValues = {
+    ...currentUser,
+    favouriteGymIds: currentUser?.favouriteGyms.map((gym) => gym.id),
+  };
+
+  const methods = useForm<EditProfileFormValues>({
+    resolver: yupResolver(UserSchema),
+    defaultValues,
+  });
+
+  const {
+    reset,
+    watch,
+    control,
+    setValue,
+    handleSubmit,
+    formState: { isSubmitting },
+  } = methods;
+
+  //Debug purposes
+  useDevWatchForm(watch);
+
+  const { runAsync: submitUploadAvatar } = useSafeRequest(uploadAvatar, {
+    manual: true,
+  });
+  const { runAsync: submitUpdateUser } = useSafeRequest(updateUser, {
+    manual: true,
+  });
+
+  const handleSubmitAvatar = async (avatar?: AvatarData) => {
+    if (avatar === undefined) {
+      return;
+    }
+
+    try {
+      const { data: uploadUrl } = await getUploadAvatarUrl();
+      await submitUploadAvatar(uploadUrl, avatar);
+    } catch (error) {
+      enqueueSnackbar('Failed to upload profile picture', { variant: 'error' });
+      throw error;
+    }
+  };
+  const handleSubmitUpdateUser = async (data: UserRequest) => {
+    try {
+      await submitUpdateUser(data);
+      enqueueSnackbar('Successfully updated profile', {
+        autoHideDuration: 5000,
+      });
+      navigate(PATH_DASHBOARD.general.jios.root);
+    } catch {
+      enqueueSnackbar('Failed to update user', { variant: 'error' });
+    }
+  };
+
+  const _handleSubmit = async ({ avatar, ...rest }: EditProfileFormValues) => {
+    try {
+      await handleSubmitAvatar(avatar);
+      await handleSubmitUpdateUser(rest);
+    } catch {
+      // Silences the error since error has been handled by each submit function
+    }
+  };
+
+  const handleClickCancelButton = () => {
+    navigate(PATH_DASHBOARD.general.profile);
+  };
+
+  // Return back to profile page if no user found.
+  // Fix for the bug where if user refreshes the page or
+  // tries to access the edit profile URL directly, currentUser will be undefined
+  useEffect(() => {
```

Comment:
i don't think currentUser can be undefined? Let's have a discussion on this

Reply:
Temporary workaround.

If a user accesses this page refreshes the Edit Profile page or tries to access the page directly via URL (instead of clicking on the 'Edit Profile" button on the Profile page), the user data only loads in after the EditProfile component is initialised, so the input fields will be empty (because the user data is undefined before it finishes loading, and defaultValues will also be undefined). 

Tried to fix with   

useEffect(() => {
  reset(defaultValues);
}, [currentUser]);

but that results in another bug as shown below:
![image](https://user-images.githubusercontent.com/62279011/197467257-2d0a8771-bca4-4e74-8d24-aa238b604b7d.png)

Hence, the workaround is to redirect users back to the profile page if the EditProfile component is initialised before the currentUser has finished loading in.

Input on this would be appreciated!

Reply:
useGetIdentity has a guard: loading, loaded and error. You can make use of those to conditionally render the component

Reply:
Fixes the issue somewhat. Thanks!